### ISSUE TITLE
Improve test coverage

### ISSUE THREAD

**Issue by abshinn** [NREL/rdtools#12]
Add unit tests for functions that have no coverage, improve existing tests. Perhaps use [the coverage module](https://coverage.readthedocs.io/en/coverage-4.3.4/) to look for lines of code that are not touched by tests.

**Comment by mdeceglie** [comment]
I just pushed a new branch, `YOY_upgrades`, which includes a change in the degradation output to tuples that emphasize the standard portion of the results. @amnag, would you have bandwidth to look at updating the tests for this change and improving the coverage for the degradation calculations? I suggest making the updates in `YOY_upgrades` branch, once this is done, we can open a pull request.

**Comment by amnag** [comment]
@mdeceglie, I have started working on it.

**Comment by amnag** [comment]
@mdeceglie , @abshinn, I have added two versions of updated code for degradation testing. These two versions are degradation_test_v1.py and degradation_test_v2.py. I think degradation_test_v2.py is cleaner and more appropriate. Can you please take a look ?

**Comment by amnag** [comment]
I just found out that I can run the updated tests directly with python but not with nosetests, trying to figure out why

**Comment by abshinn** [comment]
Hi @ambarishnag. The only reason we are using nose is because it easily searches for and runs unit tests. I just noticed that [nose is no longer maintained](http://nose.readthedocs.io/en/latest/), so perhaps it is best to no longer use it. Unittest is our main testing framework, and [it can search for tests too](https://docs.python.org/2/library/unittest.html#test-discovery).

In the meantime, to run all tests, I suggest creating a bash script that calls each test individually, for example, if you create a file `tests/run`:

```
#!/usr/bin/env bash

python -m tests.degradation_test_v2
python -m tests.normalization_pvwatts_test
...
```

**Comment by amnag** [comment]
Hi @abshinn, this has been done.

**Comment by abshinn** [comment]
Thanks, @amnag! Can you create a pull request with these changes?

**Comment by amnag** [comment]
Hi @abshinn, I created the pull request.

**Comment by abshinn** [comment]
Fixed in PR #22.
